tests/sentry/runner/test_initializer.py

```diff
@@ -187,12 +187,26 @@ def test_bootstrap_options_empty_file(settings, config_yml):
     assert settings.SENTRY_OPTIONS == {}
 
 
+def test_legacy_url_prefix_warning(settings, config_yml):
+    config_yml.write(
```

Comment:
What happens if `SENTRY_URL_PREFIX` is in `settings.py`? What if it's an envvar?

Reply:
Currently, we won't warn if `SENTRY_URL_PREFIX` is in `sentry.conf.py` (I think that's what you mean by settings.py?) or an envvar. I can add the logic for that but would like some input from the backend team since I'd lean towards changing as little logic as necessary since the logic behind this is already super messy and I don't want to add  more to it, especially since we already discourage people from using `SENTRY_URL_PREFIX`. Maybe @dashed or someone else can chime in here?

Implementation of checking user config is suggested by comment here:
https://github.com/getsentry/sentry/pull/34342#issuecomment-1121404230

Reply:
@hubertdeng123 According to `test_bootstrap_options_simple` (in this file), `config.yml` has higher precedence and should override `settings`. 

I think what @chadwhitacre is saying is that by having `settings.SENTRY_URL_PREFIX = "http://url-prefix"` here, then we expect `assert settings.SENTRY_URL_PREFIX == "http://sentry.example.com"`.

> What if it's an envvar?

I haven't seen `SENTRY_URL_PREFIX` be used as an environment variable.